### PR TITLE
Bug 1905221: pkg/cvo/sync_worker.go: Ignore work changes during init

### PR DESCRIPTION
**Pull request by jottofar**
Ignore version and overrides changes during payload initialization and log warning.

### CODE REVIEW

File: pkg/cvo/sync_worker.go [hunk 1]
```diff
@@ -476,12 +476,24 @@ func equalSyncWork(a, b *SyncWork) bool {
 	overridesEqual := reflect.DeepEqual(a.Overrides, b.Overrides)
 
 	if !sameVersion && !overridesEqual {
+		if state == payload.InitializingPayload {
+			klog.Warning("Ignoring detected version and overrides changes during payload initialization")
+			return true
+		}
 		klog.V(5).Info("Detected version and overrides changes")
```
Comment (wking) [inline pkg/cvo/sync_worker.go:483]: Trying to think of a way to DRY this up, how about:

```go
var difference string
if !sameVersion && !overridesEqual {
  difference = "version and overrides changes"
} else if ... {
  ... other cases...
}
if difference != "" {
  if state == payload.InitializingPayload {
    klog.Warningf("Ignoring detected %s during payload initialization", difference)
    return true
  }
  klog.V(5).Infof("Detected %s", difference)
  return false
}
```

Comment (jottofar) [reply pkg/cvo/sync_worker.go:483]: I can do that - and had considered it- but I think it sacrifices readability.

Comment (jottofar) [reply pkg/cvo/sync_worker.go]: Changed in [213726d](https://github.com/openshift/cluster-version-operator/pull/500/commits/213726df85663fb9ec3e7b7c5671906a3feab6a5)

**Review by wking — APPROVED** [review]
/lgtm

**Pull request merged**